ext/mysqli/mysqli_api.c

```diff
@@ -2269,10 +2325,26 @@ PHP_FUNCTION(mysqli_stmt_attr_get)
 	}
 	MYSQLI_FETCH_RESOURCE_STMT(stmt, mysql_stmt, MYSQLI_STATUS_VALID);
 
-	if ((rc = mysql_stmt_attr_get(stmt->stmt, attr, &value))) {
-		RETURN_FALSE;
+	if (attr != STMT_ATTR_CURSOR_TYPE && attr != STMT_ATTR_PREFETCH_ROWS
+#if MYSQL_VERSION_ID >= 50107
+	&& attr != STMT_ATTR_UPDATE_MAX_LENGTH
+#endif
+	) {
+		zend_argument_value_error(ERROR_ARG_POS(2), "must be one of "
+#if MYSQL_VERSION_ID >= 50107
+			"MYSQLI_STMT_ATTR_UPDATE_MAX_LENGTH, "
+#endif
+			"MYSQLI_STMT_ATTR_PREFETCH_ROWS, or STMT_ATTR_CURSOR_TYPE");
+		RETURN_THROWS();
```

Comment:
I'd feel better about moving this error into the `if()` below.

Reply:
Makes sense, I just did the new if below recently to fix the Windows issue.